src/sql/migrations/20220304184711_schema.js

```diff
@@ -1,26 +1,62 @@
-const setOnUpdateTrigger = (table) => (`
+const deleteSampleFileIfOrphanFunc = `
+CREATE OR REPLACE FUNCTION public.delete_sample_file_if_orphan()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+BEGIN
+  DELETE FROM sample_file
+  WHERE 
+    sample_file.id = OLD.sample_file_id AND
+    NOT EXISTS(
+      SELECT FROM sample_to_sample_file_map sf_map 
+      WHERE sf_map.sample_file_id = OLD.sample_file_id AND NOT OLD.sample_id = sf_map.sample_id
+    );
+  RETURN OLD;
+END;
+$function$
+`;
```

Comment:
have you tested that this triggers correctly when there is an orphaned sample and that it doesn't trigger when there is no orphaned sample?